pylhe/__init__.py

```diff
@@ -31,27 +33,94 @@ def __init__(self, **kwargs):
             raise RuntimeError
         for k,v in kwargs.iteritems():
             setattr(self,k,v)
-    
+
     @classmethod
     def fromstring(cls,string):
         obj = cls(**dict(zip(cls.fieldnames,map(float,string.split()))))
         return obj
-    
+
     def mothers(self):
         mothers = []
         first_idx  =  int(self.mother1)-1
         second_idx =  int(self.mother2)-1
         for idx in set([first_idx,second_idx]):
             if idx >= 0: mothers.append(self.event.particles[idx])
         return mothers
-    
-def loads():
+
+class LHEInit(dict):
+  """
+  Store the <init> block as dict.
+  """
+  fieldnames = ['beamA','beamB','energyA','energyB','PDFgroupA','PDFgroupB','PDFsetA','PDFsetB','weightingStrategy','numProcesses']
+
+  def __init__(self):
+    pass
+
+  @classmethod
+  def fromstring(cls,string):
+    return dict(zip(cls.fieldnames,map(float,string.split())))
+
+class LHEProcInfo(dict):
+  """
+  Store the process info block as dict.
+  """
+  fieldnames = ['xSection', 'error', 'unitWeight', 'procId']
+
+  def __init__(self):
     pass
-  
+
+  @classmethod
+  def fromstring(cls,string):
+    return dict(zip(cls.fieldnames,map(float,string.split())))
+
+def loads():
+  pass
+
 import xml.etree.ElementTree as ET
+def readLHEInit(thefile):
+    """
+    Read the init blocks. Return dict. This encodes the weight group
+    and related things according to https://arxiv.org/pdf/1405.1067.pdf
+    This function returns a dict.
+    """
+    initDict={}
+    for event,element in ET.iterparse(thefile,events=['end']):
+        if element.tag == "init":
+            data = element.text.split('\n')[1:-1]
+            initDict["initInfo"]=LHEInit.fromstring(data[0])
+            initDict["procInfo"]=[LHEProcInfo.fromstring(d) for d in data[1:]]
+        if element.tag == "initrwgt":
+            initDict["weightgroup"]={}
+            for child in element:
+                # Find all weightgroups
+                if child.tag=="weightgroup" and child.attrib!={}:
+                    try:
+                        wg_type = child.attrib["type"]
```

Comment:
I have only changed here the tag, in my lhefiles produces by MadGraph it's `name` and not `weightgroup`.